flask_appbuilder/models/sqla/interface.py

```diff
@@ -200,10 +201,15 @@ def query(
             the current page
         :param page_size:
             the current page size
+        :param base_query
+            the base SQLAlchemy Query to run to build the list view
         """
-        query = self.session.query(self.obj)
+        if base_query:
+            query = base_query(self.session)
+        else:
+            query = self.session.query(self.obj)
+            query = self._query_select_options(query, select_columns)
```

Comment:
Note: query_count is still counting the obj rows, it should count the passed query

Reply:
Good catch!